pkg/aws/actuator/actuator.go

```diff
@@ -882,12 +882,22 @@ func (a *AWSActuator) getDesiredUserPolicy(entries []minterv1.StatementEntry, us
 		Statement: []StatementEntry{},
 	}
 	for _, se := range entries {
-		policyDoc.Statement = append(policyDoc.Statement, StatementEntry{
-			Effect:    se.Effect,
-			Action:    se.Action,
-			Resource:  se.Resource,
-			Condition: se.PolicyCondition,
-		})
+		statement := StatementEntry{
+			Effect:   se.Effect,
+			Action:   se.Action,
+			Resource: se.Resource,
+		}
+		if se.PolicyCondition != nil {
+			// We're just passing through the policyCondition content right through to the AWS API
+			// so convert the rawextension content
+			var out interface{}
+			err := json.Unmarshal(se.PolicyCondition.Raw, &out)
+			if err != nil {
+				return "", fmt.Errorf("error unmarshaling PolicyCondition: %v", err)
+			}
+			statement.Condition = out
+		}
+		policyDoc.Statement = append(policyDoc.Statement, statement)
```

Comment:
Were you able to manually test with this PolicyCondition actually in use?

Reply:
Yes, with and without it. That's how I figured that I needed to annotate the StatementEntry struct to put the 'omitempty' (otherwise the JSON sent to AWS when there was no policy always included "Condition": null which would cause the AWS API calls to fail).

FWIW, this is a simple example:
```
spec:
  secretRef:
    name: my-cred-request-secret
    namespace: openshift-cloud-credential-operator
  providerSpec:
    apiVersion: cloudcredential.openshift.io/v1
    kind: AWSProviderSpec
    statementEntries:
    - effect: Allow
      action:
      - s3:CreateBucket
      - s3:DeleteBucket
      resource: "*"
      policyCondition:
        StringEquals:
          aws:userid: "testuser"
```